### ISSUE TITLE
Steam client is "forgetting" the language settings

### ISSUE THREAD

**Issue by FurretUber** [ValveSoftware/steam-for-linux#5060]
#### Your system information

* Steam client version (build number or date): 8 Jun 2017
* Distribution (e.g. Ubuntu): Xubuntu 17.04 amd64
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
I use Steam in Brazilian Portuguese, however after the computer is rebooted or Steam is reopened some times, problems start to happen with the language. It is noticeable when the problems start, because it is when the window asking to create a new Steam account appears, even if there should open the window only asking for the password.

On this particular time, the client has mixed languages: the Library is still in Brazilian Portuguese, but the Store is in English, parts of Friends window are in both idioms. In settings, the text is in Brazilian Portuguese, however in Interface is selected English but in the text with the link to translations is written: 

`Fluente em inglês e português? Ajude-nos a traduzir o Steam.`

Which is what appears when Brazilian Portuguese is the selected language.

After this particular buggy time, after I restart everything is in English.

Expected behavior: if a language is chosen, it will be always used.

Observed behavior: the language is forgotten by the Steam Client, and in one session it has English and Brazilian Portuguese mixed.

#### Steps for reproducing this issue:

1. Set Steam language to Brazilian Portuguese;
2. Restart the computer and open Steam some times;
3. Notice how the language reverts to English.

**Comment by kisak-valve** [comment]
Possibly a side effect of #5035.

**Comment by FurretUber** [comment]
The issue appears to be similar. The localization issue may be related, but to me it happens much more often than every two days. Only today this happened twice, the difference was lower than 12 hours.

**Comment by Aztorius** [comment]
Same issue with French.

1. Set language to French and restart Steam.
2. Restart the computer or reload Steam.
3. Steam may forgot username and password. And language is full English.

**Comment by darealshinji** [comment]
I remember a much older discussion about the same issue.

The Steam client overwrites `$HOME/.steam/registry.vdf` with data that contains no user info:
```
"Registry"
{
	"HKLM"
	{
		"Software"
		{
			"Valve"
			{
				"Steam"
				{
					"SteamPID"		"0"
					"TempAppCmdLine"		""
				}
			}
		}
	}
}
```

I was able to "fix" this by adding the required settings and then making the file read-only:
```
"Registry"
{
	"HKCU"
	{
		"Software"
		{
			"Valve"
			{
				"Steam"
				{
					"language"		"german"
					"AutoLoginUser"		"djcj"
					"RememberPassword"		"1"
					"SourceModInstallPath"		"/home/djcj/.local/share/Steam/steamapps/sourcemods"
					"AlreadyRetriedOfflineMode"		"0"
				}
			}
		}
	}
}
```

It's weird because sometimes after an installation all this doesn't happen. Maybe it has something to do with timestamps, but I really don't know. A side-effect of having the settings file read-only may be that, if you close Steam during a download or installation it may not continue automatically after a restart. Verifying the cache of a game can help here.

**Comment by Kiibakun** [comment]
I'm affected by this too. I hate it.

**Comment by soredake** [comment]
@Kiibakun `sudo chattr +i ~/.steam/registry.vdf`

**Comment by Kiibakun** [comment]
But that file contains not only my language configuration, it also has "installed apps" and things. Thanks for the workaround but will not have some bad side-effect?

**Comment by soredake** [comment]
@Kiibakun these `if you close Steam during a download or installation it may not continue automatically after a restart`, or try this https://github.com/ValveSoftware/steam-for-linux/issues/5030#issuecomment-325124645

**Comment by Kiibakun** [comment]
Uhm, thanks, then i'll do that, i never close steam while updating. 😄

**Comment by LucienPe** [comment]
Seems to be fixed in the Steam beta client for me.

**Comment by kolen** [comment]
Having the same problem on MacOS for a long time. Updated to beta, seems that it's fixed in it, selected language is now preserved between restarts.

**Comment by ThePierrasse** [comment]
Still having this issue on Ubuntu 16.04.3.

**Comment by SHooZ** [comment]
still relevant

**Comment by klebes** [comment]
Nenhuma soluçao, para este problema estou usando o Steam pelo debian 8, possuo dois computadores com o mesmo sistema operacional, e ambos com o mesmo problema com o Steam:perder o idioma, nao salvar login e senha e mais algumas coisinhas, até parece que não comprei um produto, para não darem  nenhuma solução com o update

**Comment by ThePierrasse** [comment]
I have the exact same issue with the only difference being that my two computers run Ubuntu 16.04.3 and not Debian 8.

**Comment by klebes** [comment]
@ThePierrasse nota-se uma falta de sensibilidade na hora de compatiblizar a Steam para o sistema Linux e olha que nos dois usamos duas distros mais utilizadas por usuarios dos sistemas Linux

**Comment by klebes** [comment]
I never again played my steam games, but I will update my steam to see if they have already solved

**Comment by FurretUber** [comment]
Since I reported this issue, I upgraded Xubuntu to 18.04 Bionic Beaver and there were a few updates on the Steam client, but this particular issue was not solved yet.

**Comment by callisto2410** [comment]
I copied the file `steam.desktop` to the directory `/home/user/.local/share/applications` and changed all the launch commands to:
`Exec=sh -c "sed -i 's/english/russian/' /home/user/.steam/registry.vdf; steam steam://open/news"`

While this works fine, even if you change the language in the settings to incorrect manually.

**Comment by anolting** [comment]
issue still exists.

**Comment by Leon422** [comment]
Issue still happens and it's extremely annoying. Steam keeps forgetting my login info and/or language randomly.

Ubuntu Budgie 18.04 64 bits

**Comment by legluondunet** [comment]
Same as Leon.

**Comment by kassindornelles** [comment]
the problem still persists.

Kubuntu 18.04.1
i always have to change the language to portuguese (brasil) after a reboot. is **really** annoying.

**Comment by sircola** [comment]
yeah, got effected by this and it's so annoying that I'm here lol. I'm on ubuntu 18.04 and I want to use hungarian always. 
Anyways I still love steam.

**Comment by YokiToki** [comment]
Still relevant (Linux Mint 19.1)

**Comment by darealshinji** [comment]
@ steam devs:
If this can be fixed by changing the access rights of the registry.vdf file, why don't you just do that as a temporary solution until you figured out the actual cause of this?
